hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto

```diff
@@ -1815,6 +1818,16 @@ message SnapshotPurgeRequest {
   repeated string updatedSnapshotDBKey = 2;
 }
 
+message SnapshotUpdateSizeRequest {
+  repeated SnapshotSize snapshotSize = 1;
+}
+
+message SnapshotSize {
```

Comment:
```suggestion
message SnapshotProperty {
```

Reply:
Actually. We can use `SnapshotInfo` directly if this is generalized into `SetProperty`

Reply:
We can generalise this, but we shouldn't send the snapshot info as it is. Imagine this case,
1) I get `SnapshotInfo` in KeyDeletetingService, the snapshot is still active.
2) Now someone deletes the snapshot, So snapshot metadata says SnapshotDeleted.
3) Now when I update the SnapshotInfo it still says it's active. Though the snapshot is deleted, we will make it active again with this flow and it will never be GCed. This is the same for all the other metadata in the snapshot as well.

Your thoughts on this?

Reply:
@aswinshakil Yup that can be a problem. But I'm worried that refactoring would be much harder later on (when we implement this and want to merge update SnapshotInfo operations).

Can we use a snapshotTableLock to synchronize the operations (e.g. snapshot delete and snapshot update), so that the case you listed above won't happen?

The current approach relies on the current state of single-threaded execution of Ratis Txs. And this is can also be a hidden problem when we enable parallel Ratis Txs at some point. So a `SNAPSHOT_LOCK` for example inside of `OMSnapshotUpdateSizeRequest#validateAndUpdateCache` would still be required. Just like the `VOLUME_LOCK/BUCKET_LOCK` that we use in volume/bucket/key operations.

Reply:
I agree we need to have snapshot lock in general for these operations. I will create another JIRA to address this issue. I was just discussing this with @hemantk-12 as well. Yes, we should not rely only on Ratis to serialize the request for the snapshot.